src/js/tour.js

```diff
@@ -315,6 +315,7 @@ export class Tour extends Evented {
 
         if (modalContainer) {
           modalContainer.remove();
+          this.modal = null
```

Comment:
I'm wondering if we need to set this to `null` or if we can just show/hide the modal?

Reply:
again not 100% sure but i think i did this to make sure we re-attach the modal element to dom since `modalContainer.remove();` removes the modal completely from the dom `show()` will do nothing (since the element is already deleted)

Reply:
Hmm, should we maybe just hide/show it instead of fully removing? It's odd that the overlay behavior is different than the behavior of the tour itself. If the tour works without calling `start` again, then we should probably make the modal do the same.

Reply:
I get your point. I didn't want to temper to much with how the whole tour works. Also after some thinking I actually came to the conclusion that keeping it the way it is/was is probably the best. If you end/cancel a tour it the modal should be removed since it's not needed anymore. If you restart (or choose to continue) the modal gets initialised again. Keeps the dom clean.

Reply:
Gotcha. Would you be able to write some tests to ensure this continues working as expected going forward? It also looks like we have some lint issues to fix.

Reply:
I think we just need a test that covers the workflow you are trying to accomplish like start tour -> advance a couple steps -> end tour -> call show on the step you left off on -> assert that the modal overlay exists and is shown.